app/views/layouts/result_content.erb

```diff
@@ -58,11 +58,40 @@
         </div>
       </div>
     </div>
+    <!-- Progress Bar -->
     <section id='content'>
+      <div class="progress_meter">
+        <span id="progress_span"><p>
+        </p></span>
+      </div>
       <%= render partial: 'layouts/noscript'%>
       <%= content_for?(:content) ? yield(:content) : yield %>
     </section>
-
+    <%
+      if @current_user.admin?
+        num_marked = @assignment.get_num_marked()
+        num_assigned = @assignment.get_num_assigned()
+      else
+        num_marked = @assignment.get_num_marked(@current_user.id)
+        num_assigned = @assignment.get_num_assigned(@current_user.id)
+      end
+    %>
+    <script type="text/javascript">
+        var marked = <%= num_marked %>
```

Comment:
I'd rather not introduce global variables here. The easiest thing to do is perform the function call by passing in `num_marked` and `num_assigned` directly, and then in the AJAX response, call `get_num_marked` and `get_num_assigned` again to recompute the numbers and pass them to `update_bar`.